lua/lualine.lua

```diff
@@ -12,9 +12,11 @@ local theme_set = {}
 M.options = {
   icons_enabled = true,
   theme = 'gruvbox',
-  separator = '|',
+	component_separators = {'', ''},
+	section_separators = {'', ''},
```

Comment:
I think it would be better to have this set up to work this way:
```lua
separator = { component = {'something', 'something'}, section = {'something', 'something'}
```
while also keeping the functionality of 
```lua
separator = '|'
```
I think the configuration will get easier if you want a static separator like it is right now and also make code in `component.lua` a lot easier.

Reply:
>I think the configuration will get easier if you want a static separator like it is right now and also make code in component.lua a lot easier.

I'm not sure how is it posible . Adding type checking to determine which type of configuration is being used will only increse complexity

Reply:
>separator = { component = {'something', 'something'}, section = {'something', 'something'}

donn't you think that's abit too nested for an option? 
lualine.options.separator.section[1] ?

I can add single string to set both to same separator but that will be processed while loading components. Adding type checking wverywhere separators are used will increse complecity...

Reply:
Okay, let's leave it for now. I will think about it. Let's fix the other issues first

Reply:
kept component_separators = '|'

Reply:
Okay, here's an idea. It is redundant to have both `separator` and `component_separators` as an option. Maybe change it to `component_separator` only? Then do `if #component_separator == 1` etc in `component.lua`

Reply:
There's no separator option.

I'm just turning the component_separations into a table with two copies of same separator if a string is given insted of a table.

I think you missunderstood